internal/nodes/ingress_model_rel.go

```diff
@@ -540,56 +541,69 @@ func parseServicesForRoute(routeSpec routev1.RouteSpec, key string) []string {
 func HostRuleToIng(hrname string, namespace string, key string) ([]string, bool) {
 	var err error
 	var oldFqdn, fqdn string
+	var fqdnType, oldFqdnType string
 	var oldFound bool
 
 	allIngresses := make([]string, 0)
 	hostrule, err := lib.AKOControlConfig().CRDInformers().HostRuleInformer.Lister().HostRules(namespace).Get(hrname)
 	if k8serrors.IsNotFound(err) {
 		utils.AviLog.Debugf("key: %s, msg: HostRule Deleted", key)
-		_, fqdn = objects.SharedCRDLister().GetHostruleToFQDNMapping(namespace + "/" + hrname)
-		if !strings.Contains(fqdn, lib.ShardVSSubstring) {
+		oldFound, oldFqdn = objects.SharedCRDLister().GetHostruleToFQDNMapping(namespace + "/" + hrname)
+		if !strings.Contains(oldFqdn, lib.ShardVSSubstring) {
 			objects.SharedCRDLister().DeleteHostruleFQDNMapping(namespace + "/" + hrname)
-			objects.SharedCRDLister().DeleteFQDNFQDNTypeMapping(fqdn)
+			oldFqdnType = objects.SharedCRDLister().GetFQDNFQDNTypeMapping(oldFqdn)
+			objects.SharedCRDLister().DeleteFQDNFQDNTypeMapping(oldFqdn)
 		}
 	} else if err != nil {
 		utils.AviLog.Errorf("key: %s, msg: Error getting hostrule: %v", key, err)
 		return nil, false
 	} else {
 		if hostrule.Status.Status != lib.StatusAccepted {
-			return allIngresses, false
+			return []string{}, false
 		}
 		fqdn = hostrule.Spec.VirtualHost.Fqdn
 		oldFound, oldFqdn = objects.SharedCRDLister().GetHostruleToFQDNMapping(namespace + "/" + hrname)
 		if oldFound && !strings.Contains(oldFqdn, lib.ShardVSSubstring) {
 			objects.SharedCRDLister().DeleteHostruleFQDNMapping(namespace + "/" + hrname)
+			oldFqdnType = objects.SharedCRDLister().GetFQDNFQDNTypeMapping(oldFqdn)
 		}
 		if !strings.Contains(fqdn, lib.ShardVSSubstring) {
 			objects.SharedCRDLister().UpdateFQDNHostruleMapping(fqdn, namespace+"/"+hrname)
-			objects.SharedCRDLister().UpdateFQDNFQDNTypeMapping(fqdn, string(hostrule.Spec.VirtualHost.FqdnType))
+			fqdnType = string(hostrule.Spec.VirtualHost.FqdnType)
+			if fqdnType == "" {
+				fqdnType = string(akov1alpha1.Exact)
+			}
+			objects.SharedCRDLister().UpdateFQDNFQDNTypeMapping(fqdn, fqdnType)
 		}
 	}
 
-	// find ingresses with host==fqdn, across all namespaces
-	ok, obj := SharedHostNameLister().GetHostPathStore(fqdn)
-	if !ok {
-		utils.AviLog.Debugf("key: %s, msg: Couldn't find hostpath info for host: %s in cache", key, fqdn)
-	} else {
-		for _, ingresses := range obj {
-			for _, ing := range ingresses {
-				if !utils.HasElem(allIngresses, ing) {
-					allIngresses = append(allIngresses, ing)
+	// in case the hostname is updated, we need to find ingresses for the old ones as well to recompute
+	if oldFound {
+		allOldFqdns := SharedHostNameLister().GetHostsFromHostPathStore(oldFqdn, oldFqdnType)
+		for _, i := range allOldFqdns {
```

Comment:
should we use 'fqdn'/ anyother name instead of `i`?

Reply:
the singular version of `allOldFqdns` -> `oldFqdn` was already taken. I thought adding another fqdn like thing might just cause confusion 😅. There are too many `*fqdn*` variables already.